addons/website_cf_turnstile/static/src/js/turnstile.js

```diff
@@ -4,52 +4,116 @@ import "@website/snippets/s_website_form/000";  // force deps
+                <script class="s_turnstile">
+                    // Rethrow the error, or we only will catch a "Script error" without any info
+                    // because of the script api.js originating from a different domain.
+                    function throwTurnstileError(code) {
+                        const error = new Error("Turnstile Error");
+                        error.code = code;
+                        throw error;
+                    }
+                    function turnstileSuccess() {
+                        let form = this.wrapper.parentElement.parentElement;
+                        let formButton = form.querySelector("button[type='submit']");
+                        let webButton  = form.querySelector("a.s_website_form_send[role='button']");
+                        if (formButton) {
+                            formButton.disabled = false;
+                            formButton.classList.remove('disabled');
+                            formButton.querySelector('i.turnstile-spinner').remove();
                         }
-                    </script>
-                    <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
-                `).insertAfter('.s_website_form_send, .o_website_form_send');
-            }
-            return res;
-        },
-
-        /**
-         * Remove potential existing loaded script/token
-         */
-        cleanTurnstile: function () {
-            if (this.$('.s_turnstile').length) {
-                this.$('.s_turnstile').remove();
-            }
-        },
-
-        /**
-         * @override
-         * Discard all library changes to reset the state of the Html.
-         */
-        destroy: function () {
+                        if (webButton) {
+                            webButton.classList.remove('disabled');
+                            webButton.querySelector('i.turnstile-spinner').remove();
+                        }
+                    }
+                    function turnstileBecomeVisible() {
+                        this.wrapper.parentElement.style.display = ''
+                    }
+                </script>
+                <script class="s_turnstile" src="https://challenges.cloudflare.com/turnstile/v0/api.js"></script>
+            `);
+        }
+    },
+
+    /**
+     * Remove potential existing loaded script/token
+     */
+    cleanTurnstile: function () {
+        if (this.$('.s_turnstile').length) {
+            this.$('.s_turnstile').remove();
+        }
+    },
+
+    /**
+     * @override
+     * Discard all library changes to reset the state of the Html.
+     */
+    destroy: function () {
+        this.cleanTurnstile();
+        this._super(...arguments);
+    },
+};
+
+const signupTurnStile = {
+    ...turnStile,
+
+    addSpinner(button) {
+        let spinner = document.createElement('i')
+        spinner.classList.add('fa', 'fa-refresh', 'fa-spin', 'turnstile-spinner')
```

Comment:
Add margin so the icon is not sticked to the button text
```suggestion
        spinner.classList.add("fa", "fa-refresh", "fa-spin", "turnstile-spinner", "me-2");
```